src/notation/internal/scorecallbacks.cpp

```diff
@@ -22,16 +22,44 @@
 #include "scorecallbacks.h"
 
 #include "libmscore/engravingitem.h"
-#include "libmscore/lyrics.h"
-#include "libmscore/system.h"
 
-#include "inotationinteraction.h"
 #include "igetscore.h"
+#include "inotationinteraction.h"
 
 #include "log.h"
 
 using namespace mu::notation;
 
+ScoreCallbacks::ScoreCallbacks(INotationInteraction* interaction, IGetScore* getScore)
+    : m_getScore(getScore), m_interaction(interaction)
+{
+    setScore(m_getScore->score());
+
+    getScore->scoreChanged().onNotify(this, [this]() {
+        setScore(m_getScore->score());
+    });
+}
+
+ScoreCallbacks::~ScoreCallbacks()
+{
+    if (m_score) {
+        m_score->removeViewer(this);
+    }
+}
+
+void ScoreCallbacks::setScore(Ms::Score* score)
```

Comment:
With this function we will have two independent ways to get a score (via m_score or via m_getScore), which is pretty confusing